### PR TITLE
[JAVA PMD] Provide Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

We provide Sider's recommended ruleset for JAVA PMD.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

None.

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Updated environments:
 Misc:
 
 - **ESLint** Re-provide new recommended configuration [#2157](https://github.com/sider/runners/pull/2150)
+- **PMD Java** Provide new recommended configuration [#2165](https://github.com/sider/runners/pull/2196)
```
Comment (suetsuguk) [inline CHANGELOG.md:18]: Wrong IDs.
```suggestion
- **PMD Java** Provide new recommended configuration [#2296](https://github.com/sider/runners/pull/2296)
```

File: images/pmd_java/sider_recommended_pmd.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    <exclude name="ClassWithOnlyPrivateConstructorsShouldBeFinal" />
+    <exclude name="CollapsibleIfStatements" />
+    <exclude name="CyclomaticComplexity" />
+    <exclude name="DataClass" />
+    <exclude name="ExcessiveImports" />
+    <exclude name="ExcessiveMethodLength" />
+    <exclude name="GodClass" />
+    <exclude name="ImmutableField" />
+    <exclude name="LawOfDemeter" />
+    <exclude name="LoosePackageCoupling" />
```
Comment (suetsuguk) [inline images/pmd_java/sider_recommended_pmd.xml:115]: Can you add the same note for `LoosePackageCoupling` as in the existing [default-ruleset.xml](https://github.com/sider/runners/blob/5200589b9947978ff70ffd76d6a7d07d49c3535e/images/pmd_java/default-ruleset.xml#L13-L14)?

```suggestion
    <!-- NOTE: Prevent the warning "No packages or classes specified" -->
    <exclude name="LoosePackageCoupling" />
```

File: CHANGELOG.md (hunk 1 above)
Comment (sleeeto) [reply CHANGELOG.md:18]: Thanks, I fixed.

File: images/pmd_java/sider_recommended_pmd.xml (hunk 2 above)
Comment (sleeeto) [reply images/pmd_java/sider_recommended_pmd.xml:115]: OK, I fixed it.

**Comment by sleeeto** [comment]
@ybiquitous Can you check them ?

**Review by ybiquitous — APPROVED** [review]

**Pull request merged**